packages/application/src/index.ts

```diff
@@ -794,33 +769,33 @@ namespace Private {
    */
   export function ensureNoCycle(
     data: IPluginData,
-    pluginMap: PluginMap,
-    serviceMap: ServiceMap
+    plugins: Map<string, IPluginData>,
+    services: Map<Token<any>, string>
   ): void {
-    let dependencies = data.requires.concat(data.optional);
+    const dependencies = data.requires.concat(data.optional);
```

Comment:
I was hesitant to activate the eslint rule for that feature. But it may be a good thing to do for 2.0.

Xref: https://eslint.org/docs/latest/rules/prefer-const

Reply:
I think this is a good idea for a targeted PR.